### PR TITLE
feat(getUsage): updates method for use with account key

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: packages/buckets/src/buckets.ts [hunk 1]
```diff
@@ -262,6 +262,13 @@ export class Buckets extends GrpcAuthentication {
     return super.getToken(identity)
   }
 
+  /**
+   * {@inheritDoc @textile/hub#GrpcAuthentication.getToken}
+   */
+  setToken(token: string) {
```
Comment (andrewxhill) [inline packages/buckets/src/buckets.ts:268]: been on my todo to add this method for a while, finally needed it

File: packages/hub/src/users.spec.ts [hunk 2]
```diff
@@ -101,16 +105,100 @@ describe('All apis...', () => {
...
-        const res = await users.getThread('my-buckets')
-        expect(res.id).to.deep.equal(users.context.toJSON()['x-textile-thread'])
+        const res = await db.getThread('my-buckets')
+        expect(res.id).to.deep.equal(db.context.toJSON()['x-textile-thread'])
 
         // The dev should see that the key was used to create one thread
         // @todo: Use the cli to list keys
       })
+
+      it('should check the users own usage', async function () {
```
Comment (andrewxhill) [inline packages/hub/src/users.spec.ts:119]: next three tests are the bulk of it

File: packages/users/src/api/index.ts [hunk 3]
```diff
@@ -94,6 +94,16 @@ export interface CustomerUsage {
...
 }
 
+/**
+ * GetUsage options
+ */
+export interface UsageOptions {
+  /**
+   * Pubkey of the user. Only available when authenticated using an account key.
+   */
+  key?: string
```
Comment (andrewxhill) [inline packages/users/src/api/index.ts:104]: this name may update before we release

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -34,6 +35,23 @@ jobs:
         image: mongo:latest
         ports:
           - '127.0.0.1:27017:27017'
+      billing:
+        image: textile/textile:billing
+        restart: always
+        environment:
```
Comment (sanderpick) [inline .github/workflows/test.yml:41]: -> `env`

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -34,6 +35,23 @@ jobs:
         image: mongo:latest
         ports:
           - '127.0.0.1:27017:27017'
+      billing:
+        image: textile/textile:billing
+        restart: always
```
Comment (sanderpick) [inline .github/workflows/test.yml:40]: Can remove, prob not a GH action thing

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -34,6 +35,23 @@ jobs:
...
+        environment:
+          - BILLING_LOG_DEBUG
+          - BILLING_LOG_FILE
```
Comment (sanderpick) [inline .github/workflows/test.yml:43]: Remove hyphens? comparing to the above hub `env` section

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -34,6 +35,23 @@ jobs:
...
+          - BILLING_ADDR_API=/ip4/0.0.0.0/tcp/10006
+          - BILLING_ADDR_MONGO_URI=mongodb://mongo:27017
+          - BILLING_ADDR_MONGO_NAME
+          - BILLING_ADDR_GATEWAY_HOST=/ip4/0.0.0.0/tcp/8010
+          - BILLING_ADDR_GATEWAY_URL
+          - BILLING_STRIPE_API_KEY=${{ secrets.STRIPE_API_KEY }}
+          - BILLING_SEGMENT_API_KEY
```
Comment (sanderpick) [inline .github/workflows/test.yml:50]: All of these keys that don't have values can be removed. They're meant to be used with an `.env` file, which is not a thing here.

File: packages/buckets/package.json [hunk 8]
```diff
@@ -30,9 +30,9 @@
   "license": "MIT",
   "dependencies": {
     "@improbable-eng/grpc-web": "^0.13.0",
-    "@textile/buckets-grpc": "2.1.7",
     "@textile/context": "^0.9.2",
     "@textile/crypto": "^1.0.1",
+    "@textile/buckets-grpc": "2.1.4",
```
Comment (sanderpick) [inline packages/buckets/package.json:35]: Downgrade?

**Review by sanderpick — COMMENTED** [review]
Awesome, that was fast. Couple nits on the CI file, otherwise LGTM!

File: packages/buckets/package.json (hunk 8 above)
Comment (andrewxhill) [reply packages/buckets/package.json:35]: not sure what happened there. npm link hiccup i think

File: packages/buckets/package.json [hunk 9]
```diff
@@ -30,16 +30,16 @@
...
-    "@textile/grpc-authentication": "^1.0.1",
-    "@textile/grpc-connection": "^1.0.1",
-    "@textile/grpc-transport": "^0.1.0",
+    "@textile/buckets-grpc": "2.1.7",
+    "@textile/grpc-authentication": "^1.0.2",
+    "@textile/grpc-connection": "^1.0.2",
+    "@textile/grpc-transport": "^0.2.1",
     "@textile/hub-grpc": "2.1.7",
-    "@textile/hub-threads-client": "^2.0.1",
+    "@textile/hub-threads-client": "^2.0.2",
```
Comment (carsonfarmer) [inline packages/buckets/package.json:40]: Glad you caught this. I have no idea why this _isn't_ updating in master? But the released version is...

File: packages/buckets/src/spec.util.ts [hunk 10]
```diff
@@ -61,12 +61,3 @@ export const signUp = (ctx: ContextInterface, addrGatewayUrl: string, sessionSec
     },
   )
 }
-
```
Comment (carsonfarmer) [inline packages/buckets/src/spec.util.ts:64]: nice

File: packages/grpc-authentication/package.json [hunk 11]
```diff
@@ -30,8 +30,8 @@
   "dependencies": {
     "@textile/context": "^0.9.2",
     "@textile/crypto": "^1.0.1",
-    "@textile/grpc-connection": "^1.0.1",
-    "@textile/hub-threads-client": "^2.0.1",
+    "@textile/grpc-connection": "^1.0.2",
```
Comment (carsonfarmer) [inline packages/grpc-authentication/package.json:33]: 🙏

File: packages/hub-threads-client/src/client.spec.ts [hunk 12]
```diff
@@ -4,8 +4,8 @@ import { SignupResponse } from '@textile/hub-grpc/api/hubd/pb/hubd_pb'
 import { expect } from 'chai'
 import { PrivateKey } from '@textile/crypto'
 import { Context } from '@textile/context'
-import { expirationError } from '@textile/security'
-import { signUp, createKey, createAPISig } from './spec.util'
+import { expirationError, createAPISig } from '@textile/security'
```
Comment (carsonfarmer) [inline packages/hub-threads-client/src/client.spec.ts:7]: perfect

File: packages/hub/src/users.spec.ts [hunk 13]
```diff
@@ -2,18 +2,21 @@ import path from 'path'
 import fs from 'fs'
 import { expect } from 'chai'
 import { isBrowser } from 'browser-or-node'
+import { grpc } from '@improbable-eng/grpc-web'
```
Comment (carsonfarmer) [inline packages/hub/src/users.spec.ts:5]: is this needed? I don't see its usage below, but I might just be missing something?

File: packages/hub/src/users.spec.ts [hunk 14]
```diff
@@ -69,27 +74,26 @@ describe('All apis...', () => {
...
-        expect(users.context.toJSON()).to.have.ownProperty('x-textile-api-sig')
+        db = new Client(userContext)
+        const token = await db.getToken(identity)
+        users = new Users(userContext)
+        users.setToken(token)
+        expect(db.context.toJSON()).to.have.ownProperty('x-textile-api-sig')
       }).timeout(3000)
 
       it('should then create a db for the bucket', async function () {
         // @todo https://github.com/textileio/js-threads/pull/263 should fix this...
```
Comment (carsonfarmer) [inline packages/hub/src/users.spec.ts:88]: I think we can remove this todo now, as that fix has been merged

File: packages/hub/src/users.spec.ts [hunk 15]
```diff
@@ -101,16 +105,100 @@ describe('All apis...', () => {
...
-        expect(res.id).to.deep.equal(users.context.toJSON()['x-textile-thread'])
+        const res = await db.getThread('my-buckets')
+        expect(res.id).to.deep.equal(db.context.toJSON()['x-textile-thread'])
 
         // The dev should see that the key was used to create one thread
         // @todo: Use the cli to list keys
       })
+
+      it('should check the users own usage', async function () {
+        if (isBrowser) return this.skip()
```
Comment (carsonfarmer) [inline packages/hub/src/users.spec.ts:120]: why not in browser? too slow?

**Review by carsonfarmer — APPROVED** [review]
Nice, this looks great @andrewxhill. I've made a few comments, mostly questions, about a few items. I don't think they need to be "changed" before merging, but maybe?

File: packages/hub/src/users.spec.ts (hunk 15 above)
Comment (andrewxhill) [reply packages/hub/src/users.spec.ts:120]: it's relying on counts of things done in node tests

File: .github/workflows/test.yml [hunk 16]
```diff
@@ -14,6 +12,26 @@ jobs:
     if: "!contains(github.event.head_commit.message, 'skip-ci')"
 
     services:
+      ipfs:
+        image: ipfs/go-ipfs:v0.5.1
```
Comment (sanderpick) [inline .github/workflows/test.yml:16]: This can be `0.7.0`

**Comment by carsonfarmer** [comment]
Hey @andrewxhill , I think we need to get this in here soon don't we?

**Pull request merged**